examples/distributed-map/locking/PessimisticUpdate.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+//
+// Created by İhsan Demir on 21/12/15.
+//
+#include <hazelcast/client/HazelcastClient.h>
+
+class Value : public hazelcast::client::serialization::IdentifiedDataSerializable {
+public:
+    int amount;
+
+    Value() {
+    }
+
+    Value(const Value &rhs) {
+        amount = rhs.amount;
+    }
+
+    bool equals(const Value &rhs) {
+        return rhs.amount == amount;
+    }
+
+    bool operator==(const Value &rhs) {
+        return equals(rhs);
+    }
+
+    int getFactoryId() const {
+        return 1;
+    }
+
+    int getClassId() const {
+        return 6;
+    }
+
+    void writeData(hazelcast::client::serialization::ObjectDataOutput &out) const {
+        out.writeInt(amount);
+    }
+
+    void readData(hazelcast::client::serialization::ObjectDataInput &in) {
+        amount = in.readInt();
+    }
+};
+
+int main() {
+    try {
+        hazelcast::client::ClientConfig config;
+        hazelcast::client::HazelcastClient hz(config);
+
+        hazelcast::client::IMap<std::string, Value> map =
+                hz.getMap<std::string, Value>("map");
+
+        std::string key("1");
+        Value v;
+        map.put(key, v);
+        std::cout << "Starting" << std::endl;
+        for (int k = 0; k < 1000; k++) {
+            map.lock(key);
+            try {
+                map.get(key);
+                hazelcast::util::sleepmillis(10);
+                v.amount++;
+                map.put(key, v);
+
+                map.unlock(key);
+            } catch (...) {
```

Comment:
I think we should not use universal catch in our examples. We should not promote it. Our users should only expect a well defined Exception. All our exceptions extends hazelcast::exception::IException, we should use that instead.

And this code block should have been as follows with `finally`

            map.lock(key);
             try {
                 map.get(key);
                 hazelcast::util::sleepmillis(10);
                 v.amount++;
                 map.put(key, v);
            } finally {
                       map.unlock(key);
             }